src/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 configure_file(version.h.in version.h)
 
+# Add compiler warnings
+if (MSVC)
+	add_compile_options(/W4)
+else()
+	add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
```

Comment:
How many warnings are there to address. Is it eventually feasible to convert the warnings to errors for the CI?